pkg/client/cli/cloud_messages.go

```diff
@@ -0,0 +1,157 @@
+	if err != nil {
+		return &systema.CommandMessageResponse{}, err
+	}
+
+	systemaClient := systema.NewSystemACliClient(conn)
+	return systemaClient.GetUnauthenticatedCommandMessages(ctx, &empty.Empty{})
+}
+
+func (cmc *cloudMessageCache) updateCacheMessages(ctx context.Context, resp *systema.CommandMessageResponse) {
+	// Update the messages
+	cmc.Intercept = resp.GetIntercept()
+
+	// Update the time to do the next check since we were successful
+	refreshMsgs := client.GetConfig(ctx).Cloud.RefreshMessages
+	cmc.NextCheck = dtime.Now().Add(refreshMsgs)
+
+	// We reset the messages delivered for all commands since they
+	// may have changed
+	for key := range cmc.MessagesDelivered {
+		cmc.MessagesDelivered[key] = false
+	}
+}
+
+func (cmc *cloudMessageCache) getMessageFromCache(ctx context.Context, cmdUsed string) string {
+	// Ensure that the message hasn't already been delivered to the user
+	// if it has, then we don't want to print any output so as to not
+	// annoy the user.
+	var msg string
+	if val, ok := cmc.MessagesDelivered[cmdUsed]; ok {
+		if val {
+			return msg
+		}
+	}
+
+	// Check if we have a message for the given command
+	switch cmdUsed {
+	case "intercept":
+		msg = cmc.Intercept
+	default:
+		// We don't currently have any messages for this command
+		// so our msg will remain empty
+	}
+	cmc.MessagesDelivered[cmdUsed] = true
+	return msg
+}
+
+// raiseCloudMessage is what is called from `PostRunE` in a command and is responsible
+// for raising the message for the command used.
+func raiseCloudMessage(cmd *cobra.Command, _ []string) error {
+	ctx := cmd.Context()
+	// Currently we only have messages that should be served when a user
+	// isn't logged in, so we check that here
+	if cliutil.HasLoggedIn(cmd.Context()) {
+		if _, err := cliutil.GetCloudUserInfo(ctx, false, true); err == nil {
+			return nil
+		}
+	}
+
+	// If the user has specified they are in an air-gapped cluster,
+	// we shouldn't try to get messages
+	if client.GetConfig(cmd.Context()).Cloud.SkipLogin {
+		return nil
+	}
+
+	// The command is the first word of cmd.Use
+	cmdUsed := strings.Split(cmd.Use, " ")[0]
+
+	// Load the config
+	cmc, err := newCloudMessageCache(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Check if it is time to get new messages from Ambassador Cloud
+	if dtime.Now().After(cmc.NextCheck) {
+		env, err := client.LoadEnv(ctx)
+		if err != nil {
+			return err
+		}
+		systemaURL := fmt.Sprintf("https://%s:%s", env.SystemAHost, env.SystemAPort)
+		resp, err := getCloudMessages(ctx, systemaURL)
+		if err != nil {
+			// We try again in an hour since we encountered an error
+			cmc.NextCheck = dtime.Now().Add(1 * time.Hour)
+		} else {
+			cmc.updateCacheMessages(ctx, resp)
+		}
+	}
+
+	// Get the message from the cache that should be delivered to the user
+	msg := cmc.getMessageFromCache(ctx, cmdUsed)
+	fmt.Fprintf(cmd.OutOrStdout(), "%s\n", msg)
```

Comment:
nit: maybe only print `if msg != ""`? otherwise this'll print a newline every time